ursh/indico_ursh/controllers.py

```diff
@@ -26,26 +26,38 @@
 from indico.web.rh import RH
 
 from indico_ursh.util import request_short_url
+from indico_ursh.views import WPShortenURLPage
 
 
-class RHShortenURL(RH):
+class RHGetShortURL(RH):
     """Make a request to the URL shortening service"""
 
     @staticmethod
-    def _resolve_full_url(original_url):
-        if url_parse(original_url).host is not None:
+    def _resolve_full_url(original_url, require_absolute):
+        if url_parse(original_url).host:
             return original_url
+        if require_absolute:
+            raise BadRequest('Invalid URL for URL shortening service')
         original_url = original_url.lstrip('/')
         return posixpath.join(config.BASE_URL, original_url)
 
     @staticmethod
     def _check_host(full_url):
+        print(full_url, url_parse(full_url).host)
         if url_parse(full_url).host != url_parse(config.BASE_URL).host:
             raise BadRequest('Invalid host for URL shortening service')
 
     def _process(self):
         original_url = request.json.get('original_url')
-        full_url = self._resolve_full_url(original_url)
+        require_absolute = request.json.get('absolute')
```

Comment:
why not just check client-side in JS if it's absolute? since the value here is client-provided anyway there's no benefit in doing it on the server